CYCLONEDDS_QUALITY_DECLARATION.md

```diff
@@ -100,27 +125,42 @@ In all, the test code appears to comprise approximately 25% of the codebase.
 
 ### Feature Testing [4.i]
 
+Each feature in `CycloneDDS` has corresponding tests which simulate typical usage, and they are located in separate directories next to the sources. New features are required to have tests before being added.
+Currently nightly results can be seen here:
+* [linux-aarch64_release](https://ci.ros2.org/view/nightly/job/nightly_linux-aarch64_release/lastBuild/testReport/rmw_cyclonedds_cpp/)
+* [linux_release](https://ci.ros2.org/view/nightly/job/nightly_linux_release/lastBuild/testReport/rmw_cyclonedds_cpp/)
+* [mac_osx_release](https://ci.ros2.org/view/nightly/job/nightly_osx_release/lastBuild/testReport/rmw_cyclonedds_cpp/)
+* [windows_release](https://ci.ros2.org/view/nightly/job/nightly_win_rel/lastBuild/testReport/rmw_cyclonedds_cpp/)
+
 A substantial amount of the tests found throughout the source tree appear to verify functionality of various features of `CycloneDDS`.
 However the lack of a complete feature list (see section [3.i]) makes it difficult to analyze the breadth of the tests.
 
 ### Public API Testing [4.ii]
 
+Each part of the public API has tests, and new additions or changes to the public API require tests before being added. The tests aim to cover both typical usage and corner cases, but are quantified by contributing to code coverage.
 There are some tests throughout the `CycloneDDS` source tree which specifically target the public API, but there are no policies or mechanisms to discover and measure which tests target the public API or how much of the public API is covered by the existing tests.
 
 ### Coverage [4.iii]
 
-There is no test coverage tracking in `CycloneDDS`.
+There is no test coverage tracking in `CycloneDDS`. Automated coverage runs are being added to the CI to show coverage.
+
 
 ### Performance [4.iv]
 
+`CycloneDDS` performance is verifiably good and regression free per ROS Tooling WG's [nightly CI performance tests](http://build.ros2.org/job/Fci__nightly-performance_ubuntu_focal_amd64/ ).
```

Comment:
It's less important to state that the performance is strong, and it's probably close to impossible to say that performance is regression free. Instead, in this section, we just want to see that performance is tracked so that performance regressions can be resolved. Performance regressions don't even need to be fixed if the code that introduced them is reasonable to have. Just so long as they are found as part of the development process and appropriate resolutions identified.

How are these nightly jobs incorporated into cyclonedds development?

Reply:
@brawner our experience this past year is that Open Robotics CI testing infrastructure and tests are not reliable. More hours are spent debugging OR CI testing infrastructure and tests than "actual" cyclonedds issues. Not sure if the OR CI can be a source of measurement for reliably ascertaining quality and performance. We are very eager for OR CI to be improved. I've brought this up a few times in ROS Tooling WG.